src/components/Slider/Slider.tsx

```diff
@@ -8,46 +8,65 @@ import SliderMark from './components/SliderMark';
 import SliderThumb from './components/SliderThumb';
 import SliderTrack from './components/SliderTrack';
 import { Container, InputContainer, InputsContainer } from './Slider.style';
+import { IMarkProps, IThumbProps, ITrackProps } from 'react-range/lib/types';
 
-interface Props {
-  /** Determines the position of the rendered thumbs. Defaults to [0, 100] */
-  values: number[];
+type CommonProps = {
   /** Determines if the Slider is disabled or not */
   disabled?: boolean;
-  /** Determines whether the Slider is a Range or a Selector */
-  isSelector?: boolean;
   /** Determines if the Slider will show increments along its track */
   hasIncrements?: boolean;
-  /** Use this to change the slider's initial values (e.g. [80, 100]) */
-  initialValues?: number[];
   /** Called when a thumb is moved, provides new values */
   onChange: (values: number[]) => void;
   /** Called when a change is finished (mouse/touch up, or keyup), provides current values.
    * Use this event when you have to make for example ajax request with new values. */
   onFinalChange?: (values: number[]) => void;
-}
+};
+
+type SelectorProps = {
+  /** Determines the position of the rendered thumbs and the type of the Slider. 1 value means it's a Selector,
+   * while 2 values mean it's a Range. Defaults to [0, 100] */
+  values: [number];
+};
+
+type RangeProps = {
+  /** Determines the position of the rendered thumbs and the type of the Slider. 1 value means it's a Selector,
+   * while 2 values mean it's a Range. Defaults to [0, 100] */
+  values: [number, number];
+};
```

Comment:
Since this has been simplified, maybe now you could just have a single `Props` with a 

```
/** Determines the position of the rendered thumbs and the type of the Slider. 1 value means it's a Selector,
   * while 2 values mean it's a Range. Defaults to [0, 100] */
values: [number] | [number, number];
```